src/checks/#cc4a#modern_language.clas.abap

```diff
@@ -0,0 +1,1157 @@
+          code_line = ')'.
+        ELSE.
+          code_line = ''.
+        ENDIF.
+        quickfix->replace(
+           context = assistant_factory->create_quickfix_context(
+              VALUE #( procedure_id = procedure-id
+                       statements = VALUE #( from = statement_index to = statement_index )
+                       tokens = VALUE #( from = receiving_idx to = to_idx ) ) )
+           code = VALUE #( ( code_line ) ) ).
+        method_line = |{ result_line } = { method_line }|.
+      ENDIF.
+
+      quickfix->replace(
+        context = assistant_factory->create_quickfix_context(
+           VALUE #( procedure_id = procedure-id
+                    statements = VALUE #( from = statement_index to = statement_index )
+                    tokens = VALUE #( from = 1 to = 3 ) ) )
+        code = VALUE #( ( method_line ) ) ).
+
+      IF to_idx <> lines( statement-tokens ) AND method_name NP '*('.
+        quickfix->insert_after(
+          context = assistant_factory->create_quickfix_context(
+             VALUE #( procedure_id = procedure-id
+                      statements = VALUE #( from = statement_index to = statement_index )
+                      tokens = VALUE #( from = lines( statement-tokens ) to = lines( statement-tokens ) ) ) )
+          code = VALUE #( (  `)` ) ) ).
+      ENDIF.
+    ENDIF.
+    add_finding(
+    EXPORTING
+       procedure = procedure
+       statement_index = statement_index
+       code = c_code_call_method
+       pseudo_comment = c_ps_call_method
+       quickfixes = quickfixes
+    CHANGING findings = findings ).
+  ENDMETHOD.
+
+
+  METHOD analyze_exporting_receiving.
+    DATA exporting_idxs TYPE SORTED TABLE OF i WITH UNIQUE KEY table_line.
+    DATA(statement) = procedure-statements[ statement_index ].
+    IF analyzer->find_clause_index( tokens = statement-tokens clause = 'EXCEPTIONS' ) <> 0
+    OR analyzer->find_clause_index( tokens = statement-tokens clause = 'IMPORTING' ) <> 0
+    OR analyzer->find_clause_index( tokens = statement-tokens clause = 'CHANGING' ) <> 0.
+      RETURN.
+    ENDIF.
+    LOOP AT statement-tokens ASSIGNING FIELD-SYMBOL(<token>) WHERE references IS INITIAL AND lexeme = 'EXPORTING'.
+      DATA(token_idx) = sy-tabix.
+      IF check_remove_exporting( statement = statement token_idx = token_idx ) = abap_true.
+        INSERT token_idx INTO TABLE exporting_idxs.
+      ENDIF.
+    ENDLOOP.
+
+    get_receiving_infos( EXPORTING tokens = statement-tokens
+                         IMPORTING receiving_idx = DATA(receiving_idx)
+                                   end_idx       = DATA(end_idx)
+                                   result_line   = DATA(result_line) ).
+
+    IF result_line IS INITIAL AND exporting_idxs IS INITIAL.
+      RETURN.
+    ENDIF.
+    DATA(quickfixes) = assistant_factory->create_quickfixes( ).
+    IF result_line IS NOT INITIAL.
+      DATA(finding_code) = c_code_exporting_receiving.
+      DATA(ps) = c_ps_exporting_receiving.
```

Comment:
Pspsps? I guess this is supposed to be `pseudo_comment` and not a method of attracting cats? ;)